sdk/storage/storage-queue/src/QueueClient.ts

```diff
@@ -313,6 +493,39 @@ export class QueueClient extends StorageClient {
     super(url, pipeline);
     this._queueName = this.getQueueNameFromUrl();
     this.queueContext = new Queue(this.storageClientContext);
+
+    // Declare MessagesContext
+    // Build the url with "messages"
+    const partsOfUrl = this.url.split("?");
+    const messagesUrl = partsOfUrl[1]
+      ? appendToURLPath(partsOfUrl[0], "messages") + "?" + partsOfUrl[1]
+      : appendToURLPath(partsOfUrl[0], "messages");
+    // Duplicating the following lines from StorageClient constructor
+    let storageClientContext = new StorageClientContext(
+      messagesUrl,
+      this.pipeline.toServiceClientOptions()
+    );
+    // Override protocol layer's default content-type
+    (storageClientContext as any).requestContentType = undefined;
```

Comment:
extract common code to a helper that takes url and pipeline then returns a storage client context

Reply:
Done.